workflows/funcscan.nf

```diff
@@ -79,43 +67,117 @@ workflow FUNCSCAN {
 
     ch_input = Channel.fromSamplesheet("input")
 
+    ///////////////////////
+    // INPUT PREPARATION //
+    ///////////////////////
+
     // Some tools require uncompressed input
-    fasta_prep = ch_input
-        .branch {
-            compressed: it[1].toString().endsWith('.gz')
-            uncompressed: it[1]
-        }
+    ch_input_prep = ch_input
+                        .map{meta, fasta, faa, gbk -> [meta, [fasta, faa, gbk]]}
+                        .transpose()
+                        .branch {
+                            compressed: it[1].toString().endsWith('.gz')
+                            uncompressed: it[1]
+                        }
 
-    GUNZIP_FASTA_PREP ( fasta_prep.compressed )
-    ch_versions = ch_versions.mix( GUNZIP_FASTA_PREP.out.versions )
+    GUNZIP_INPUT_PREP ( ch_input_prep.compressed )
+    ch_versions = ch_versions.mix( GUNZIP_INPUT_PREP.out.versions )
 
     // Merge all the already uncompressed and newly compressed FASTAs here into
     // a single input channel for downstream
-    ch_unzipped_fastas = GUNZIP_FASTA_PREP.out.gunzip
-                        .mix( fasta_prep.uncompressed )
+    ch_intermediate_input = GUNZIP_INPUT_PREP.out.gunzip
+                            .mix( ch_input_prep.uncompressed )
+                            .groupTuple()
+                            .map{
+                                meta, files ->
+                                    def fasta_found   = files.find{it.toString().tokenize('.').last().matches('fasta|fas|fna|fa')}
+                                    def faa_found     = files.find{it.toString().endsWith('.faa')}
+                                    def gbk_found     = files.find{it.toString().tokenize('.').last().matches('gbk')}
```

Comment:
Should also add 'gbk|gbff' also as that is the gbk extension output for `bakta`.

Reply:
Ill add it here as i couldnt comment on it for some reason :D 
So in line 188 
```
    // TODO: Only NT at the moment. AA tax. classification will be added only when its PR is merged.
```
remove this because now that the user supllies always both fasta and gbk files for the preannotated track, we dont need to update the taxonomy workflow ;)

Reply:
Oooh that's a good question.. currently only FASTAs go to taxonomy... I can't remember if that's preanno ones too

Reply:
Wait, the user doesn't always supply GBK, I don't understand now...?